olive/resource_path.py

```diff
@@ -34,7 +35,11 @@ def __str__(self) -> str:
 
 
 LOCAL_RESOURCE_TYPES = [ResourceType.LocalFile, ResourceType.LocalFolder]
-AZUREML_RESOURCE_TYPES = [ResourceType.AzureMLModel, ResourceType.AzureMLDatastore, ResourceType.AzureMLJobOutput]
+AZUREML_RESOURCE_TYPES = [
+    ResourceType.AzureMLModel,
+    ResourceType.AzureMLDatastore,
+    ResourceType.AzureMLJobOutput,
```

Comment:
Is there a reason `AzureMLRegistryModel` is not added here?

Reply:
We are doing a couple of validations based on aml resource type like workspace validation, and if aml client exist validation for those aml user workspace assets. Those validations looks more like for user-aml-workspace. AML registry model basically doesn't need those validation. It is just a cloud storage place to download the model. So instead of updating existing validation logic, I removed it from aml resource type.

Reply:
Thanks! Makes sense to me. 
Can you update this mapping https://github.com/microsoft/Olive/blob/main/olive/systems/azureml/aml_system.py#L41 so that aml model can support aml registry model? 
the new type just needs to map to custom_model.

Reply:
Will update this. Question: after preparing non-local model (download it), the resource type won't be updated to local resource right?

Reply:
The preparation only happens for non-aml system, so if both host and target are aml, it won't be downloaded.